core/src/solution_submission/retry.rs

```diff
@@ -39,7 +41,11 @@ impl<'a> InfallibleGasPriceEstimator<'a> {
     async fn estimate(&mut self) -> U256 {
         match self.gas_price_estimating.estimate_gas_price().await {
             Ok(gas_estimate) => {
-                self.previous_gas_price = gas_estimate.fast;
+                let price = gas_estimate.fast;
```

Comment:
nit: Any reason not to use `cmp::max`?
```suggestion
                let price = cmp::max(gas_estimate.fast, self.previous_gas_price);
```